src/sql/azdata.proposed.d.ts

```diff
@@ -1801,5 +1828,69 @@ declare module 'azdata' {
 
 		export interface Dialog extends LoadingComponentBase {
 		}
+
+		/**
+		 * Provides dialog options to customize modal dialog content and layout
+		 */
+		export interface ICustomDialogOptions {
+			/**
+			 * Severity Level to identify icon of modal dialog.
+			 */
+			severity: MessageLevel;
```

Comment:
We should add a name property here too like in `createWizard` and other similar ones to use to identify in telemetry events (name is non-localized non-PII field identify the type of dialog it is)

Reply:
As of current design, these options only apply for ErrorMessageDialog, which is only used for Connection errors.. but we may be able to extend it to 'name' the dialog in case other paths want to consume the same dialog in future. 

Since we don't have a use-case as of now, I'll follow up on this separately, probably with the remaining Diagnostic RP design.